packages/plugin-avatar/src/avatar-url-store.js

```diff
@@ -0,0 +1,104 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {patterns} from '@ciscospark/common';
+
+/**
+ * <uuid+size, {uuid, size, url}> map
+ */
+const urlByUuid = new WeakMap();
+
+/**
+ * @class AvatarUrlStore
+ */
+export default class AvatarUrlStore {
+  /**
+   * @constructs {AvatarUrlStore}
+   */
+  constructor() {
+    urlByUuid.set(this, new Map());
+  }
+
+  /**
+   * Get the URL associated with the given uuid and size.
+   *
+   * @param {object} item
+   * @param {string} item.uuid A user uuid
+   * @param {integer}item.size the requested size
+   * @returns {Promise} resolves to the URL or rejects if not mapped
+   *
+   * @memberOf AvatarUrlStore
+   */
+  get(item) {
+    if (!item) {
+      return Promise.reject(new Error(`\`item\` is required`));
+    }
+    if (!item.uuid) {
+      return Promise.reject(new Error(`\`item.uuid\` is required`));
+    }
+    if (!item.size) {
+      return Promise.reject(new Error(`\`item.size\` is required`));
+    }
+    if (!patterns.uuid.test(item.uuid)) {
+      return Promise.reject(new Error(`\`item.uuid\` does not appear to be a uuid`));
+    }
+
+    const ret = urlByUuid.get(this).get(`$(item.uuid) - $(item.size)`);
+    if (ret) {
+      return Promise.resolve(ret.url);
+    }
+    return Promise.reject(new Error(`No URL found by specified id`));
+  }
+
+  /**
+   * Adds the given item to the store
+   * @param {object} item
+   * @param {string} item.uuid
+   * @param {integer} item.size
+   * @param {string} item.url
+   * @returns {Promise<item>}
+   */
+  add(item) {
+    if (!item) {
+      return Promise.reject(new Error(`\`item\` is required`));
+    }
+    if (!item.uuid) {
+      return Promise.reject(new Error(`\`uuid\` is required`));
+    }
+
+    if (!item.size) {
+      return Promise.reject(new Error(`\`size\` is required`));
+    }
+
+    if (!patterns.uuid.test(item.uuid)) {
+      return Promise.reject(new Error(`\`uuid\` does not appear to be a uuid`));
+    }
+
+    if (!item.url) {
+      return Promise.reject(new Error(`\`url\` is required`));
+    }
+
+    setTimeout(this.remove.bind(this, item), this.config.cacheExpiration);
+    urlByUuid.get(this).set(`$(item.uuid) - $(item.size)`, item);
```

Comment:
The avatar service returns a cacheControll specifying the TTL of the url. I used the existing implementation but I wonder if we shouldn't be using the server suggested value (which is exactly the same as our configuration)